deploy/helm/elastic-agent/templates/agent/eck/statefulset.yaml

```
@@ -37,7 +38,7 @@ spec:
     replicas: {{ . }}
     {{- end }}
     podTemplate:
-{{ include "elasticagent.engine.eck.podTemplate" (list $ $presetVal $agentName) }}
+{{- toYaml ($podTemplateResource).template | nindent 4 }}
```

```suggestion
{{- toYaml ($podTemplateResource).template | nindent 6 }}
```
`spec` for eck should be under `podTemplate`